File: Data Structures & Algorithms/invert-a-binary-tree/submission-0.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    public TreeNode invertTree(TreeNode root) {
        // if root is empty then no children
        if (root == null) return null;

        // swap left child with right child
        TreeNode temp = root.left;
        root.left = root.right;
        root.right = temp;

        // recursively keep swapping
        invertTree(root.left);
        invertTree(root.right);

        // once you reach root return
        return root;

    }
}

File: Data Structures & Algorithms/lowest-common-ancestor-in-binary-search-tree/submission-0.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def lowestCommonAncestor(self, root: TreeNode, p: TreeNode, q: TreeNode) -> TreeNode:
        # if root val is greater than p and greater than q, go left
        if root.val > p.val and root.val > q.val:
            return self.lowestCommonAncestor(root.left, p, q)
        # if root val is greater than p and greater than q, go right
        if root.val < p.val and root.val < q.val:
            return self.lowestCommonAncestor(root.right, p, q)
        # if left<=root<=right OR left>=root>=right we found the LCA
        if root.val >= p.val and root.val <= q.val or root.val <= p.val and root.val >= q.val:
            return root